### ISSUE TITLE
Amulet of avarice bonus applied incorrectly.

### ISSUE THREAD

**Issue by EnuDelrum** [weirdgloop/osrs-dps-calc#370]
### What went wrong?

At least when it comes to magic damage, the Amulet of avarice bonus against revenants it being incorrectly applied after the magic damage bonus instead of being added with the magic damage bonus.

For example, putting this in the DPS calc (with Fire Surge and 99 Magic) you'd expect a max hit of 33.
https://dps.osrs.wiki?id=SlacksFiveLumber
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/128336524/24b47191-7786-4713-ae3c-de3156337a5d)

In-game it actually hits for 32.
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/128336524/bf250fb2-900f-4787-b629-71b74544f146)

Example of what's happening:
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/128336524/c4cdde90-7c7f-47f8-9457-2d9618ba2342)

And (for what it's worth) this is how it fits into all of the magic damage max hit calculations (not including non-pvp tomes):
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/128336524/e7bedd17-b910-4f7c-907a-59acc2b2271b)

### What browsers are you seeing the problem on?

Chrome

### What device(s) are you seeing the problem on?

Windows

### Any other information

It seems like Forinthry Surge is correct as it's just changing the avarice's bonus from 20% to 35%.

**Comment by jmyaeger** [comment]
Just FYI, your magic max hit model is slightly inaccurate. Ahrim's + amulet of the damned is a post-roll effect that occurs after the mark of darkness boost, and the tomes were changed to be multiplied in at the end of the pre-roll calculation rather than post-roll.

Also, re: accursed sceptre's interaction with the slayer boost, have you tested that in-game to confirm that the two boosts are additive? They're implemented as multiplicative (slayer first, then  sceptre) in the calc at the moment, so it would be good to know if that's wrong.

**Comment by EnuDelrum** [comment]
@jmyaeger Thank you for the info. I know my tomes were still wrong since the update. But I spent the weekend doing some thorough testing. You were right about Ahrim's + amulet of the damned being post-roll. It was hard for me to verify since there's no indication if you're getting a higher hit without the effect or a lower hit with the effect triggering. Testing this with mark of darkness, I looked at if they're additive or multiplicative along with which is multiplied first. The numbers match the mark of darkness and then the 30% amulet bonus (when it rolls) being multiplicative in that order. And of course, both are happening post-roll.
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/128336524/02b41ba0-bf43-4dee-912b-721ac741f63d)

You're also correct that after the initial magic damage bonus (such as gear, prayer, etc) that slayer, thammaron's/accursed sceptre and tomes are each multiplicative in that order. Accursed sceptre is also multiplicative with its special attack.
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/128336524/8d4e85fc-46e6-4184-9283-cf410d51dc17)

![image](https://github.com/weirdgloop/osrs-dps-calc/assets/128336524/fcaa6408-bebc-4444-9f41-5d7ca5f4006b)

I checked all of these on the DPS calc. Besides this bug report for the amulet of avarice, everything else matched up with the in-game numbers. I've updated my spreadsheet too.
![image](https://github.com/weirdgloop/osrs-dps-calc/assets/128336524/32444071-db71-4e50-a7bb-24c6fbc3de9c)
